File: mars/src/main/java/TurnLeft.java
/**
 * @author guang
 * @since 2019-08-21 16:51
 */
public class TurnLeft extends Turn {
    @Override
    public Point execute(Point point) {
        switch (point.getDirection()) {
            case SOUTH:
                point.setDirection(Direction.WEST);
                break;
            case WEST:
                point.setDirection(Direction.NORTH);
                break;
            case NORTH:
                point.setDirection(Direction.EAST);
                break;
            case EAST:
                point.setDirection(Direction.SOUTH);
                break;
            default:
        }
        return point;
    }
}

File: mars-java-kotlin/src/main/kotlin/TurnLeft.kt
/**
 * 左转指令
 */
class TurnLeft : Turn() {
    open override fun excute(point: Point): Point {

        when (point.direction) {
            Direction.SOUTH -> point.direction = Direction.WEST
            Direction.WEST -> point.direction = Direction.NORTH
            Direction.NORTH -> point.direction = Direction.EAST
            Direction.EAST -> point.direction = Direction.SOUTH
            else -> throw RuntimeException("point.Direction 不合法$point")
        }

        return point
    }
}